pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -654,6 +658,10 @@ func (lrw *logicalReplicationWriterProcessor) flushBuffer(
 		return nil, 0, err
 	}
 
+	if err := ctx.Err(); err != nil {
```

Comment:
>  each worker

It is vanishingly unlikely, but imagine that the scheduler hasn't run the loop in awhile so it hasn't kicked off new workers, but has run all the prior workers until they exited, then the loop finally gets scheduled and sees the ctx is cancelled and breaks. No worker ever hit an error, so g.Wait has no worker error to return, but we didn't do all the work -- since we broke out of the work dispatching loop -- so we need to return an error all the same.

You might ask why we don't just return the error in the loop rather than breaking, but returning without waiting for Wait() is how you leak goroutines, thus we break+wait+check.

Reply:
Added an inline comment to this effect.